lib/common/util/exceptions.js

```diff
@@ -0,0 +1,130 @@
+// @flow
+
+import moment from 'moment'
+
+import type {ParsedRangesAndDates} from '../../types'
+
+// Method to filter out new dates (0's) so that flow is happy
+// We have to use a reducer to keep flow happy.
+export function filterOutNewDates (dates: Array<string | 0>): Array<string> {
+  const outputDates: Array<string> = []
+  const filter = (acc, current) => {
+    typeof current === 'string' && acc.push(current)
+    return acc
+  }
+  dates.reduce(filter, outputDates)
+  return outputDates
+}
+
+/**
+ * This method adds a range of dates from startRange to endRange to the dates array, inclusively according to startInclusive
+ * and endInclusive.
+ */
+export function addRangeOfDates (
+  startRange: moment,
```

Comment:
Is `moment` a valid flow type?

Reply:
Yep, I was basing this off of [this](https://stackoverflow.com/questions/43036634/how-to-specify-a-moment-js-object-annotation-in-flow).